### PR TITLE
@alloy => Set initial filter params from URL on overview tab

### PR DESCRIPTION
**Pull request by mzikherman**
This parses any initial query params and injects it into the current filter state.

It also removes state from `Radio` and `Checkbox`.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 0f20e0a079003d8932a931c59fdf827e4466113a

https://deploy-preview-1029--artsy-reaction.netlify.com

File: src/Apps/Artist/Routes/Overview/state.ts [hunk 1]
```diff
@@ -19,6 +19,23 @@ export class FilterState extends Container<State> {
     sort: "-partner_updated_at",
   }
 
+  constructor(props: any) {
```
Comment (alloy) [inline src/Apps/Artist/Routes/Overview/state.ts]: Should the input not just be `state: State` ?

File: src/Apps/Artist/routes.tsx [hunk 2]
```diff
@@ -47,8 +49,20 @@ export const routes = [
...
+
+          return (
+            <Provider inject={[new FilterState(props.location.query)]}>
+              <Component artist={props.artist} />
+            </Provider>
+          )
+        },
         prepareVariables: (params, props) => {
-          return { ...props.location.query, ...params }
+          const initialFilterState = props.location ? props.location.query : {}
```
Comment (alloy) [inline src/Apps/Artist/routes.tsx]: Can you add a `FIXME` comment that notes the weird behaviour which is probably a bug?

File: src/Styleguide/Elements/Checkbox.tsx [hunk 3]
```diff
@@ -24,41 +24,13 @@ export interface CheckboxProps {
   onSelect?: any
```
Comment (alloy) [inline src/Styleguide/Elements/Checkbox.tsx]: `(selected: boolean) => void`

**Comment by alloy** [comment]
Nice! Just a few small comments, feel free to merge after addressing as you see fit.

**Comment by alloy** [comment]
Oops

File: src/Apps/Artist/Routes/Overview/state.ts (hunk 1 above)
Comment (mzikherman) [reply src/Apps/Artist/Routes/Overview/state.ts]: Ah of course.

File: src/Styleguide/Elements/Checkbox.tsx (hunk 3 above)
Comment (mzikherman) [reply src/Styleguide/Elements/Checkbox.tsx]: Yea, I can do that in `Radio` too.

**Pull request merged**